File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>set;
        set.insert(nums.begin(),nums.end());
        int n=nums.size();
        int maxCount=0;
        for(int ele:set){
            // int ele=nums[i];
            int count=1;
            if(set.find(ele-1)==set.end()){
                while(set.find(ele+1)!=set.end()){
                    ele++;
                    count++;
                }
            }
            maxCount=max(maxCount,count);
        }
        return maxCount;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer>set=new HashSet<>();
        int n=nums.length;
        for(int i=0;i<n;i++){
            set.add(nums[i]);
        }
        int maxLen=0;
        for(int count:set){
            
            if(!set.contains(count-1)){
               int len=1;
               count++;
               while(set.contains(count)){
                  len++;
                  count++;
               }
               maxLen=Math.max(maxLen,len);
            }
        }
        return maxLen;
    }
}